### PR TITLE
configurable k value

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal

### CODE REVIEW

File: node/hyperbahn/egress-nodes.js [hunk 1]
```diff
@@ -59,6 +59,13 @@ EgressNodes.prototype.kValueFor = function kValueFor(serviceName) {
         self.defaultKValue;
 };
 
+EgressNodes.prototype.setDefaultKValue = function setDefaultKValue(kValue) {
+    var self = this;
+    if (typeof kValue === 'number' && kValue > 0 && self.defaultKValue !== kValue) {
```
Comment (Raynos) [inline node/hyperbahn/egress-nodes.js]: no if check. do an `assert()` if you must.

Comment (ShanniLi) [inline node/hyperbahn/egress-nodes.js]: If someone mess up with the config file, or the config file is corrupted, assert will cause uncaught exceptions. It should be avoided.

Comment (jcorbin) [inline node/hyperbahn/egress-nodes.js]: exactly, that's called a programmer error, break fast, break hard, break early, use an assert

**Comment by jcorbin** [comment]
This PR seems incomplete?  You add a new method but don't use it?  Context in comments, docs, or something more please.

**Comment by Raynos** [comment]
It's used in rt/hyperbahn as part of flipr default k value.

**Comment by Raynos** [comment]
lgtm.

**Comment by Raynos** [comment]
Actually lets add a test.

**Pull request merged**